### PR TITLE
pocketbase: init at 0.4.2

### PR DESCRIPTION
**Pull request by dit7ya**
###### Description of changes

PocketBase is an open source Go backend, consisting of:

    embedded database (SQLite) with realtime subscriptions
    built-in files and users management
    convenient Admin dashboard UI
    and simple REST-ish API

https://github.com/pocketbase/pocketbase

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/web/pocketbase/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  vendorSha256 = "sha256-8IiY/gjK8m2ntOXyG84HMiyT4GK3CgDTRG1DB+v0jAs=";
+
+  # This is the released subpackage from upstream repo
+  subPackages = [ "examples/base" ];
+
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
```
Comment (onny) [inline pkgs/development/web/pocketbase/default.nix:24]: Could you please add a comment what does these flags mean?

Comment (dit7ya) [reply pkgs/development/web/pocketbase/default.nix:24]: Seems like all these flags are pretty common around go packages in nixpkgs? Also, they come from the upstream build instructions anyway. 

I can write the explanations if you still think that is necessary.

Comment (onny) [reply pkgs/development/web/pocketbase/default.nix:24]: Oh if there are from the upstream build instructions then you could just add a comment with a link to it :)

File: pkgs/development/web/pocketbase/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+buildGoModule rec {
```
Comment (SuperSandro2000) [inline pkgs/development/web/pocketbase/default.nix:5]:
```suggestion
}:

buildGoModule rec {
```

**Comment by SuperSandro2000** [comment]
@ofborg build pocketbase

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
You are missing a top-level entry.

**Pull request merged**